### PR TITLE
BNB signing working

### PR DESCRIPTION
**Pull request by BitHighlander**

### CODE REVIEW

File: packages/hdwallet-native/src/binance.ts [hunk 1]
```diff
@@ -77,20 +77,21 @@ export function MixinNativeBinanceWallet<TBase extends core.Constructor<NativeHD
...
+          throw Error("Invalid permissions to sign for address")
+        }
         const addressTo = msg.tx.msgs[0].outputs[0].address;
         const amount = msg.tx.msgs[0].inputs[0].coins[0].amount;
         const asset = "BNB";
         const memo = msg.tx.memo;
-        const sequence = msg.sequence;
 
-        const result = await client.transfer(addressFrom, addressTo, amount, asset, memo, sequence);
+        const result:any = await client.transfer(addressFrom, addressTo, amount, asset, memo, null);
```
Comment (BitHighlander) [inline packages/hdwallet-native/src/binance.ts:94]: Will fast follow and bring in better typeing asap

File: packages/hdwallet-native/src/binance.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import * as core from "@shapeshiftoss/hdwallet-core";
 import { BIP32Interface } from "bitcoinjs-lib";
-import BncClient from "bnb-javascript-sdk-nobroadcast";
+import {BncClient} from "bnb-javascript-sdk-nobroadcast";
```
Comment (majorhayes) [inline packages/hdwallet-native/src/binance.ts:3]: style
```suggestion
import { BncClient } from "bnb-javascript-sdk-nobroadcast";
```

File: packages/hdwallet-native/src/binance.ts [hunk 3]
```diff
@@ -77,20 +77,21 @@ export function MixinNativeBinanceWallet<TBase extends core.Constructor<NativeHD
         const { privateKey } = util.getKeyPair(this.#wallet, msg.addressNList, "binance");
 
         const client = new BncClient("https://dex.binance.org"); // broadcast not used but available
-        client.chainId = msg.chain_id;
-        client.setAccountNumber(Number(msg.account_number) || undefined);
         await client.chooseNetwork("mainnet");
-        await client.setPrivateKey(privateKey, Number.isInteger(Number(msg.account_number)));
+        await client.setPrivateKey(privateKey);
```
Comment (majorhayes) [inline packages/hdwallet-native/src/binance.ts:81]: Kevin made a PR specifically to add the account number to fix the issue of signatures being different. Can you explain how this fixes the current issue by reverting that change?

File: packages/hdwallet-native/src/binance.ts [hunk 4]
```diff
@@ -77,20 +77,21 @@ export function MixinNativeBinanceWallet<TBase extends core.Constructor<NativeHD
...
         await client.initChain();
 
         const addressFrom = msg.tx.msgs[0].inputs[0].address;
+        const addressFromVerify = client.getClientKeyAddress()
+        if(addressFrom !== addressFromVerify){
```
Comment (majorhayes) [inline packages/hdwallet-native/src/binance.ts:86]:
```suggestion
        if (addressFrom !== addressFromVerify) {
```

File: packages/hdwallet-native/src/binance.ts [hunk 5]
```diff
@@ -77,20 +77,21 @@ export function MixinNativeBinanceWallet<TBase extends core.Constructor<NativeHD
...
 
         const client = new BncClient("https://dex.binance.org"); // broadcast not used but available
-        client.chainId = msg.chain_id;
-        client.setAccountNumber(Number(msg.account_number) || undefined);
         await client.chooseNetwork("mainnet");
-        await client.setPrivateKey(privateKey, Number.isInteger(Number(msg.account_number)));
+        await client.setPrivateKey(privateKey);
         await client.initChain();
 
         const addressFrom = msg.tx.msgs[0].inputs[0].address;
```
Comment (majorhayes) [inline packages/hdwallet-native/src/binance.ts:84]:
```suggestion
        const addressFrom = msg.tx?.msgs?.[0]?..inputs?.[0]?.address;
```

**Review by majorhayes — APPROVED** [review]
I assume this fixes the issue, but I don't understand why it does.

Please fix the two style suggestions

File: packages/hdwallet-native/src/binance.ts (hunk 3 above)
Comment (BitHighlander) [reply packages/hdwallet-native/src/binance.ts:81]: this offloads the account lookup to the sdk, but perhaps is not needed. was simply trying to rule out issues debugging

Comment (BitHighlander) [reply packages/hdwallet-native/src/binance.ts:81]: ill add this back

**Pull request merged**